project_half_3.py

```diff
@@ -13,27 +15,46 @@ def printMenu():
     print("8. Save all data")
     print("9. Exit")
 
-def saveNewEntry(dict_by_id, list_of_id):
-    id = input("ID: ")
+def saveNewEntry(dict_by_id, list_of_person):
 
-    id = checkParamIsNumber(id, "ID")
+    id = checkParamIsNumber(input("ID: "), "ID")
     if id == -1:
-        return -1
+        return 0
 
     if id in dict_by_id:
         print("Error: ID already exists: " + str(dict_by_id[id]))
-        return -1
+        return 0
 
     name = input("Name: ")
 
-    age = (input("Age: ")) 
-
-    age = checkParamIsNumber(age, "age")
+    age = checkParamIsNumber(input("Age: "), "age")
     if age == -1:
-        return -1
+        return 0
     
-    dict_by_id[id] = (name, age)     
-    list_of_id.append(id)
+    flag = None
+    while flag != "1" and flag !="2":
+        flag  = input("If you are an employee - press 1, if you are a student - press 2: ")
+
+    if flag == "1":
+        field = input("Field of work: ")
+
+        salary = checkParamIsNumber(input("Salary: "), "salary",)
+        if salary == -1:
+            return 0
+        
+        person = Employee(id, name, age, field, salary)

```

Comment:
כדי שהחתימות של הקלאסים שלך יהיו זהות, וגם כדי שהפונקציה saveNewEntry תהיה קצרה יותר, אני רוצה שתכניס את 
field and salary
אל תוך ה __init__ של Employee, אותו דבר אצל סטודנט.
עכשיו זה עלול ליצור לך בעיה, כי סלרי הוא מספר - לכן תוכל להכניס את הפונקציה checkParamIsNumber אל תוך utils.py ותייבא אותו גם מהקלאסים.
ככה יהיה יותר קצר.
תצטרך לחשוב מה אתה עושה במצב שבו מה שהמשתמש הכניס הוא לא מספר - כי מהאיניט לא תוכל להחזיר 0 או מינוס אחד. אני ממליץ על לזרוק אקספשן ולתפוס אותו כאן